### ISSUE TITLE
doubled macro in dbt models

### ISSUE THREAD

**Issue by adam-tokarski** [sqlfluff/sqlfluff#7149]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

In specific situations, dbt macro calls (`{{ ... }}`) in sql files are repeated. Sometimes in the same line, sometimes in new one.

I found similar issue (https://github.com/sqlfluff/sqlfluff/issues/4367) but it was done 2 years ago, so it's either regression or something new but with similar symptoms.

### Expected Behaviour

One macro call is enough in most cases :)

### Observed Behaviour

It may be dangerous as, when applied in bigger file, one may not notice that there is doubled macro as it says that's fixed. Usually it should fails when macro is doubled, but I can imagine situation where sqlfluff is fine, rendered sql is also working, but results are broken.

### How to reproduce

Let's have such macro
```sql
{%- macro my_macro_with_somewhat_long_name() -%}
    1 = 1
{%- endmacro -%}
```

used in model
```sql
select 1 as col_1
from my_table
where {{ my_macro_with_somewhat_long_name() }}
```
in such configuration where macro call is extending line length (here, for this exercise, I've set this as 40):
```
[sqlfluff]
max_line_length = 40
```

When sqlfluff fix is applied, the model file gets the macro call repeated:
```sql
select 1 as col_1
from my_table
where
    {{ my_macro_with_somewhat_long_name() }}
{{ my_macro_with_somewhat_long_name() }}
```
and `LT05` is raised:
```
=== [dbt templater] Project Compiled.
== [dbt_local/models/my_model.sql] FAIL
L:   3 | P:   1 | LT05 | Line is too long (46 > 40). [layout.long_lines]
== [dbt_local/models/my_model.sql] FIXED
1 fixable linting violations found
```

### Dialect

trino but also saw that in postgres

### Version

```             
sqlfluff 3.4.2
sqlfluff-templater-dbt 3.4.2
Python 3.12.3
```

### Configuration

```
[sqlfluff]
dialect = trino
templater = dbt
max_line_length = 40
```

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
